cmd/csi-node-driver-registrar/node_register.go

```diff
@@ -36,23 +36,32 @@ func nodeRegister(
 	// pluginswatcher infrastructure. Node labeling is done by kubelet's csi code.
 	registrar := newRegistrationServer(csiDriverName, *kubeletRegistrationPath, supportedVersions)
 	socketPath := fmt.Sprintf("/registration/%s-reg.sock", csiDriverName)
-	fi, err := os.Stat(socketPath)
-	if err == nil && (fi.Mode()&os.ModeSocket) != 0 {
-		// Remove any socket, stale or not, but fall through for other files
-		if err := os.Remove(socketPath); err != nil {
-			klog.Errorf("failed to remove stale socket %s with error: %+v", socketPath, err)
-			os.Exit(1)
-		}
-	}
-	if err != nil && !os.IsNotExist(err) {
-		klog.Errorf("failed to stat the socket %s with error: %+v", socketPath, err)
-		os.Exit(1)
-	}
 
 	var oldmask int
 	if runtime.GOOS == "linux" {
```

Comment:
Instead of the `if` here, how about wrapping up the logic in a function along the lines of `cleanupSocketFile()` and placing that in https://github.com/kubernetes-csi/node-driver-registrar/blob/master/cmd/csi-node-driver-registrar/util_linux.go and https://github.com/kubernetes-csi/node-driver-registrar/blob/master/cmd/csi-node-driver-registrar/util_windows.go ?

Reply:
makes sense. Will update the PR

Reply:
Updated the PR. PTAL.

Reply:
I think Deep's intention was to not have to expose an OS check here. You could have a function where the Linux implementation calls umask() and the Windows implementation does nothing

Reply:
For the umask will follow up with a short PR removing the checks, to separate out changes to existing code unrelated to this PR's context.